File: nvidia_tutorials/stage/stage_tutorial.py
from pxr import Usd, Sdf 
import os 

stage : Usd.Stage = Usd.Stage.CreateNew("_assets/root_layer_example.usda")

root_layer : Sdf.Layer = stage.GetRootLayer()
print("Root Layer identifier:", os.path.relpath(root_layer.identifier))
stage.DefinePrim("/World", "Xform")

extra_layer : Sdf.Layer = Sdf.Layer.CreateNew("_assets/extra_layer.usdc")

rel_path = "./" + os.path.basename(extra_layer.identifier)
print(f"the rel_path{rel_path}")
root_layer.subLayerPaths.append(rel_path)
stage.Save()
extra_layer.Save()

# Print the contents of the root layer:
print("Root layer contents:")
print(root_layer.ExportToString())

File: nvidia_tutorials/stage/C++/stage_tutorial.cpp
#include <pxr/usd/usd/stage.h>
#include <pxr/usd/usd/prim.h>
#include <pxr/usd/sdf/layer.h>
#include <filesystem>
#include <iostream>


using namespace pxr;
namespace fs = std::filesystem;

int main()
{
  UsdStageRefPtr stage = UsdStage::CreateNew("_assets/root_layer_example_cpp.usda");
  SdfLayerHandle rootLayer = stage->GetRootLayer();
  std::cout << "Root Layer Identifier :" 
            << fs::relative(rootLayer->GetIdentifier()) << std::endl;
  
  stage->DefinePrim(SdfPath("/World"), TfToken("Xform"));
  SdfLayerRefPtr extraLayer = SdfLayer::CreateNew("_assets/extra_layer_cpp.usdc");
  const std::string relativePath = "./" + fs::path(extraLayer->GetIdentifier()).filename().generic_string();
  rootLayer->GetSubLayerPaths().push_back(relativePath);
  stage->Save();
  extraLayer->Save();

  std::string rootLayerContents;
  rootLayer->ExportToString(&rootLayerContents);
  std::cout << "Root Layer Contents :\n" << rootLayerContents;
  return 0;
}
